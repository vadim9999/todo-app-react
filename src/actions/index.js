import { ADD_TASK, FOUND_WORD } from "../constants/action-types";


// export function addArticle(payload){
//     return { type: ADD_ARTICLE, payload}
// }


export function addTask(payload){
    console.log("Call Function in actions");
    console.log(payload);
    
    return { type: ADD_TASK, payload}
}

export function addFoundWord(payload){
    console.log("Found word");
    console.log(payload);
    
    return { type: FOUND_WORD, payload}
}

export function getData(){
    return fetch("https://jsonplaceholder.typicode.com/posts").then(
        response => response.json())
        .then(json => {
            return {
                type:"Data_LOADED",
                payload:json
            }
        })
    
}