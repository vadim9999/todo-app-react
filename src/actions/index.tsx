import { ADD_TASK, FOUND_WORD } from '../constants/action-types';

// export function addArticle(payload){
//     return { type: ADD_ARTICLE, payload}
// }

export function addTask(payload: any) {
  return { type: ADD_TASK, payload };
}

export function addFoundWord(payload: any) {
  return { type: FOUND_WORD, payload };
}

export const getTasks = (payload: any) => ({ type: 'GET_TASKS', payload });

export const deleteTaskById = (payload: any) => ({
  type: 'DELETE_TASK',
  payload
});

export const updateTaskById = (payload: any) => ({
  type: 'UPDATE_TASK',
  payload
});
// export function getData() {
//     return function(dispatch) {
//       return fetch("https://jsonplaceholder.typicode.com/posts")
//         .then(response => response.json())
//         .then(json => {
//             console.log("data");
//             console.log(json);

//           dispatch({ type: "DATA_LOADED", payload: json });
//         });
//     };
//   }

export const sortTasksByGrowthDate = (payload: any) => ({
  type: 'SORT_TASKS_BY_GROWTH_DATE',
  payload
});

export const sortTasksByDecreaseDate = (payload: any) => ({
  type: 'SORT_TASKS_BY_DECREASE_DATE',
  payload
});

export const addCurrentPage = (payload: any) => ({
  type: 'ADD_CURRENT_PAGE',
  payload
});

export const addSelectedRowKeys = (payload:any) =>({
  type:'ADD_SELECTED_ROW_KEYS',
  payload
})
export const addCurrentSelectedRowKeys = (payload: any) => ({
  type: 'ADD_CURRENT_SELECTED_ROW_KEYS',
  payload
});
// export function saveTask(){
//   return { type: "SAVE_TASK"}
// }
