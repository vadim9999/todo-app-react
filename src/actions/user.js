import { ADD_TASK, FOUND_WORD } from '../constants/action-types';

export const addUser = (payload) => ({
  type: 'ADD_USER',
  payload,
});

export const authenticate = (payload) => ({
  type: 'AUTHENTICATE', payload,
});

export const authorizate = (payload) => ({
  type: 'AUTHORIZATE', payload,
});
