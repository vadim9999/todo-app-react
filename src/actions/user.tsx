// import { ADD_TASK, FOUND_WORD } from '../constants/action-types';

export const addUser = (payload:any) => ({
  type: 'ADD_USER',
  payload,
});

export const authenticate = (payload:any) => ({
  type: 'AUTHENTICATE', payload,
});

export const authorizate = (payload:any) => ({
  type: 'AUTHORIZATE', payload,
});
