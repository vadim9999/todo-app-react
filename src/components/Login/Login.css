
.login-block{
    /* border: 2px solid red; */

    display:flex;
    flex-direction:column;
    align-items: center;
    height: 70vh;
    justify-content: center;
    /* justify-content: stretch; */
}
/* .login-form{
    /* border: 2px solid green; */
    /* display: flex; */
    /* flex-direction: column; */
    /* align-items:center; */
    
/* }  */
/* .btn-login{
    align-self: flex-end;
    margin:5px 5px 0px 0px;
    
    
} */
.btn-login:hover{
    animation: loginRotation 1s linear 1
}

@keyframes loginRotation{
    0%{
        transform:translate(0px, 0px)
    }
    25%{
        transform:translate(20px, 0px)
    }
    50%{
        transform: translate(-20px, 0px)
    }
    100%{
        transform:translate( 0px,0px)
    }
}
label{
    color: rgba(247, 233, 233, 0.733);
    align-self: flex-start;
}

.input-email,.input-pass{
    
    height: 30px;
    
    
}


.input-email{
    
    /* background-size:200% 200%; */
    /* animation: inputAnimationEmail 4s ease infinite; */
}



