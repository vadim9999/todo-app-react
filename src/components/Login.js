import React, {Component} from "react";
import {connect} from "react-redux"
import { addUser } from "../actions/user"

function mapDispatchToProps(dispatch){
    return {addUser: () => dispatch(addUser)}
}

class ConnectedLogin extends Component{
    constructor(){
        super()

    }

    onSubmit(e){
        e.preventDefault()
        console.log("ddd");
        
    }
    componentDidMount(){
        console.log("This is componentDidMount");
        
    }

    render(){
        return (
            <div>
                
                    <input type="email" name="email" />
                    <input type="password" />
                    <input type="date" name="bday" />
                    <button type="submit" onClick={this.onSubmit}>Save</button>
            </div>
        )
    }
}

const Login = connect(null, mapDispatchToProps)(ConnectedLogin)
export default Login;