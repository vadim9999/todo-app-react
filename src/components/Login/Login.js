import React, { Component } from "react";
import { Redirect } from "react-router-dom"
import { connect } from "react-redux"
import { authorizate } from "../../actions/user"
// import "./Login.css"

import { Input, Tooltip, Icon, Button } from 'antd';

const mapStateToProps = (state) => {
    return {
        user_id: state.user._id
    }
}
const mapDispatchToProps = (dispatch) => {
    return {
        authorizate: (data) => dispatch(authorizate(data)),

    }
}

class ConnectedLogin extends Component {
    constructor() {
        super()
        this.state = {
            email: "",
            password: ""
        }
        this.onHandleChange = this.onHandleChange.bind(this)
        this.login = this.login.bind(this)

    }
    login(e) {
        e.preventDefault()
        console.log("Authenticate");
        const { email, password } = this.state;
        this.props.authorizate({
            email, password
        })
        this.setState({
            email: "",
            password: "",

        })
    }
    onHandleChange(e) {
        this.setState({
            [e.target.name]: e.target.value
        })
    }
    render() {
        // console.log("Login page");

        if (this.props.user_id) return <Redirect to={"/todolist"} />

        return (

            <div className="login-block">
                <form className="login-form" onSubmit={this.login}>

                    <Input
                        placeholder="Enter your username"
                        className="input-email"
                        onChange={this.onHandleChange} 
                        value={this.state.email}
                        name = "email"
                        prefix={<Icon type="user" style={{ color: 'rgba(0,0,0,.25)' }} />}
                        suffix={
                            <Tooltip title="Extra information">
                                <Icon type="info-circle" style={{ color: 'rgba(0,0,0,.45)' }} />
                            </Tooltip>
                        }
                    />
                    <Input.Password name ="password" className="input-pass" onChange={this.onHandleChange} value={this.state.password} placeholder="input password" />

                   
                  
                    {/* <input  type="password" name="password"></input> */}
                    <Button type="primary" >Login</Button>
                    <button className="btn-login" type="submit">Login</button>
                </form>
            </div>

        )
    }
}

const Login = connect(mapStateToProps, mapDispatchToProps)(ConnectedLogin)
export default Login