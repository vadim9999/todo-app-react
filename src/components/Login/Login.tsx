import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import {
  Input, Tooltip, Icon, Button, Form, Row, Col,
} from 'antd';
import { authorizate } from '../../actions/user';
// import "./Login.css"


const mapStateToProps = (state:any) => ({
  user_id: state.user._id,
});

const mapDispatchToProps = (dispatch:any) => ({
  authorizate: (data:any) => dispatch(authorizate(data)),

});

interface LoginProps {

}
class ConnectedLogin extends Component<> {
  constructor(props:any) {
    super(props);
    this.state = {
      email: '',
      password: '',
    };
    this.onHandleChange = this.onHandleChange.bind(this);
    this.login = this.login.bind(this);
  }

  login(e:any) {
    e.preventDefault();
    console.log('Authenticate');
    const { email, password } = this.state;
    this.props.authorizate({
      email, password,
    });
    this.setState({
      email: '',
      password: '',

    });
  }

  onHandleChange(e) {
    this.setState({
      [e.target.name]: e.target.value,
    });
  }

  render() {
    // console.log("Login page");

    if (this.props.user_id) return <Redirect to="/todolist" />;

    return (

      <div>


        <Row type="flex" justify="center">
          <Col>
            <Form onSubmit={this.login}>
              <Form.Item>
                <Input
                  placeholder="Enter your username"
                  className="input-email"
                  onChange={this.onHandleChange}
                  value={this.state.email}
                  name="email"
                  prefix={<Icon type="user" style={{ color: 'rgba(0,0,0,.25)' }} />}
                  suffix={(
                    <Tooltip title="Extra information">
                      <Icon type="info-circle" style={{ color: 'rgba(0,0,0,.45)' }} />
                    </Tooltip>
                          )}
                />
              </Form.Item>
              <Form.Item>
                <Input.Password
                  name="password"
                  className="input-pass"
                  onChange={this.onHandleChange}
                  value={this.state.password}
                  placeholder="Input password"
                />

              </Form.Item>
              <Form.Item>
                <Button type="primary" htmlType="submit" className="login-form-button">Log in</Button>
              </Form.Item>
            </Form>
          </Col>

        </Row>
      </div>


    );
  }
}

const Login = connect(mapStateToProps, mapDispatchToProps)(ConnectedLogin);
export default Login;
