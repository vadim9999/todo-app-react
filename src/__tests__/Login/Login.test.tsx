import React from 'react'
import Enzyme,{mount, shallow } from 'enzyme';
import sinon from 'sinon'
import Login,{ConnectedLogin} from '../../components/Login/Login'
import configureStore from 'redux-mock-store'
import Adapter from 'enzyme-adapter-react-16'
import {Provider} from 'react-redux'
import {Form, Input} from 'antd'
import { BrowserRouter, Route, Switch, Link, Redirect } from 'react-router-dom';
import {createStore} from 'redux'
import renderer from 'react-test-renderer'
import {authorizate, addUser} from '../../actions/user'
import rootReducer from '../../reducers/index'
Enzyme.configure({adapter: new Adapter() })

describe('testing component Login', () =>{
  const initialState = {
    user: {},
    tasks: [],
    currentPage: 1,
    selectedRowKeys: [],
    currentSelectedRowKeys: []
  }

  // const mockStore = configureStore()


  let store:any, wrapper:any

  beforeEach(() =>{
    store =createStore(rootReducer,initialState)
    // store = mockStore(initialState)

    // container = shallow(
    //   <BrowserRouter>
    //   <Switch>
    //   <Login store={store} />
    //   </Switch>
    //   </BrowserRouter>
      
    //   )
    wrapper = mount(<Provider store={store} >
      <BrowserRouter>
      <Switch>
      <Login  />
      </Switch>
      </BrowserRouter>
      
      </Provider>)
  })
  
  it('test', ()=>{
    const my = wrapper.find(ConnectedLogin).prop('user_id')
    console.log(my)
     
    expect (wrapper.find(ConnectedLogin).length).toEqual(1)
    expect(wrapper.find(ConnectedLogin).prop('user_id')).toEqual(initialState.user._id)
  })

  // it('check action on dispatchin', () =>{
  //   let action 
  //   store.dispatch(authorizate({
  //     email:'3434',
  //     password:'3434'
  //   }))

  //   action = store.getActions()

  //   console.log(action);

  //   expect(action[0].type).toBe("AUTHORIZATE")

   
  // })

  it('testing form on submit when user is not logged with empty fields',()=>{
    // console.log("props in component", wrapper.find(ConnectedLogin).props() );
    // console.log(wrapper.find(Redirect).props());
    
    const LoginForm = wrapper.find(Form)
      
    LoginForm.props().onSubmit({preventDefault: jest.fn()})

    // let action = store.getActions()
    // console.log(action);
    
    // const my = props.authorizate.mock.calls[0][0].email;
    // console.log(my);
    
  })

  it('testing onTyping email in Login input',()=>{
    const InputLogin = wrapper.find(Input).at(0)

    InputLogin.props()
    .onChange({target:{value:'newemail',
  name:'email'}})

    expect(wrapper.find(ConnectedLogin).state('email')).toEqual('newemail')
    // InputLogin
    // .simulate('change',{target:{value:'addd'}})
    // expect()
    // .simulate('change',{target:{value:'My new value'}})
    
  })

  it('testing redirect to /todolist when user is logged',()=>{
    store.dispatch({type:'ADD_USER_SUCCESS',payload:{
      _id:'2445'
    }})
    wrapper.update()
    expect(wrapper.find(Redirect).prop('to')).toEqual('/todolist')
  })
  // it('calls',() =>{
  //     console.log(container.prop('user'));
      
  //     expect(container.length).toEqual(1)
  //     // expect(container.prop('user')).toEqual(initialState.user)
  //   // sinon.spy(ConnectedLogin.prototype, 'componentDidMount')
  //   // const wrapper = mount(<ConnectedLogin />)
  // })

})


describe('snapshot', () =>{
  it('snpa1',()=>{
    const renderedValue = renderer.create(<ConnectedLogin />).toJSON()
    expect(renderedValue).toMatchSnapshot()
  } )
})
 
