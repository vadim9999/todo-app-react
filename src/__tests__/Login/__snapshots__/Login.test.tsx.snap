// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot snpa1 1`] = `
<div>
  <div
    className="ant-row-flex ant-row-flex-center"
    style={Object {}}
  >
    <div
      className="ant-col"
      style={Object {}}
    >
      <form
        className="ant-form ant-form-horizontal"
        onSubmit={[Function]}
      >
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-control-wrapper"
            style={Object {}}
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <span
                  className="input-email ant-input-affix-wrapper"
                >
                  <span
                    className="ant-input-prefix"
                  >
                    <i
                      aria-label="icon: user"
                      className="anticon anticon-user"
                      style={
                        Object {
                          "color": "rgba(0,0,0,.25)",
                        }
                      }
                    >
                      <svg
                        aria-hidden="true"
                        className=""
                        data-icon="user"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                        />
                      </svg>
                    </i>
                  </span>
                  <input
                    className="ant-input"
                    name="email"
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Enter your email"
                    style={null}
                    type="text"
                    value=""
                  />
                  <span
                    className="ant-input-suffix"
                  >
                    <i
                      aria-label="icon: info-circle"
                      className="anticon anticon-info-circle"
                      onMouseEnter={[Function]}
                      onMouseLeave={[Function]}
                      style={
                        Object {
                          "color": "rgba(0,0,0,.45)",
                        }
                      }
                    >
                      <svg
                        aria-hidden="true"
                        className=""
                        data-icon="info-circle"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
                        />
                        <path
                          d="M464 336a48 48 0 1 0 96 0 48 48 0 1 0-96 0zm72 112h-48c-4.4 0-8 3.6-8 8v272c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V456c0-4.4-3.6-8-8-8z"
                        />
                      </svg>
                    </i>
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-control-wrapper"
            style={Object {}}
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <span
                  className="ant-input-password input-pass ant-input-affix-wrapper"
                >
                  <input
                    action="click"
                    className="ant-input"
                    name="password"
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Input password"
                    style={null}
                    type="password"
                    value=""
                  />
                  <span
                    className="ant-input-suffix"
                  >
                    <i
                      aria-label="icon: eye-invisible"
                      className="anticon anticon-eye-invisible ant-input-password-icon"
                      onClick={[Function]}
                      onMouseDown={[Function]}
                      tabIndex={-1}
                    >
                      <svg
                        aria-hidden="true"
                        className=""
                        data-icon="eye-invisible"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M942.2 486.2Q889.47 375.11 816.7 305l-50.88 50.88C807.31 395.53 843.45 447.4 874.7 512 791.5 684.2 673.4 766 512 766q-72.67 0-133.87-22.38L323 798.75Q408 838 512 838q288.3 0 430.2-300.3a60.29 60.29 0 0 0 0-51.5zm-63.57-320.64L836 122.88a8 8 0 0 0-11.32 0L715.31 232.2Q624.86 186 512 186q-288.3 0-430.2 300.3a60.3 60.3 0 0 0 0 51.5q56.69 119.4 136.5 191.41L112.48 835a8 8 0 0 0 0 11.31L155.17 889a8 8 0 0 0 11.31 0l712.15-712.12a8 8 0 0 0 0-11.32zM149.3 512C232.6 339.8 350.7 258 512 258c54.54 0 104.13 9.36 149.12 28.39l-70.3 70.3a176 176 0 0 0-238.13 238.13l-83.42 83.42C223.1 637.49 183.3 582.28 149.3 512zm246.7 0a112.11 112.11 0 0 1 146.2-106.69L401.31 546.2A112 112 0 0 1 396 512z"
                        />
                        <path
                          d="M508 624c-3.46 0-6.87-.16-10.25-.47l-52.82 52.82a176.09 176.09 0 0 0 227.42-227.42l-52.82 52.82c.31 3.38.47 6.79.47 10.25a111.94 111.94 0 0 1-112 112z"
                        />
                      </svg>
                    </i>
                  </span>
                </span>
              </span>
            </div>
          </div>
        </div>
        <div
          className="ant-row ant-form-item"
          style={Object {}}
        >
          <div
            className="ant-col ant-form-item-control-wrapper"
            style={Object {}}
          >
            <div
              className="ant-form-item-control"
            >
              <span
                className="ant-form-item-children"
              >
                <button
                  className="ant-btn login-form-button ant-btn-primary"
                  onClick={[Function]}
                  type="submit"
                >
                  <span>
                    Log in
                  </span>
                </button>
              </span>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
